cmd/metacache-walk.go

```diff
@@ -92,6 +85,27 @@ func (s *xlStorage) WalkDir(ctx context.Context, opts WalkDirOptions, wr io.Writ
 	}
 	defer close(out)
 
+	// Fast exit track to check if we are listing an object with
+	// a trailing slash, this will avoid to list the object content.
+	if HasSuffix(opts.BaseDir, SlashSeparator) {
+		metadata, err := ioutil.ReadFile(pathJoin(volumeDir,
+			opts.BaseDir[:len(opts.BaseDir)-1]+globalDirSuffix,
+			xlStorageFormatFile))
+		if err == nil {
+			// if baseDir is already a directory object, consider it
+			// as part of the list call, this is a AWS S3 specific
+			// behavior.
+			out <- metaCacheEntry{
```

Comment:
Will this sort correctly if there are objects within? Things sent to `out` must be in sorted order.

Reply:
yes @klauspost the top level is always in the highest order